src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = newNode;
+            tail = head;
+        } else {
+            tail.next = newNode;
+            tail = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nodeOnIndex = find(index);
+        Node<T> newNode = new Node<>(nodeOnIndex.prev, value, nodeOnIndex);
+        if (nodeOnIndex == head) {
+            head.prev = newNode;
+            head = newNode;
+        } else {
+            nodeOnIndex.prev.next = newNode;
+            nodeOnIndex.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        list.forEach(this::add);
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return find(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = find(index);
+        T oldValue = node.value;
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> nodeOnIndex = find(index);
+        unlink(nodeOnIndex);
+        return nodeOnIndex.value;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> currentNode = head;
+        for (int i = 0; i < size; i++) {
+            if ((currentNode.value == null && object == null)
+                    || (currentNode.value != null && currentNode.value.equals(object))) {
```

Comment:
```suggestion
            if (currentNode.value == object 
                    || currentNode.value != null && currentNode.value.equals(object)) {
```
this brackets are redundant (read about [Operator Precedence](https://www.programiz.com/java-programming/operator-precedence))